### ISSUE TITLE
Native Binaries in gradle distribution are not signed

### ISSUE THREAD

**Issue by ThadHouse** [gradle/gradle#14912]
We have an app that includes a build setup, which we use gradle for. We're trying to ship this app on macOS, which because of new Apple requirements must be notarized. In order for this to happen, all native binaries included in our app, even in zip form, must be signed. While working on this, we noticed that none of the native binaries in the gradle distribution bin are signed. This causes our app to fail notarization. Our only solution currently is to extract the distribution and sign the binaries manually, but this isn't a fantastic long term solution. It would be helpful if gradle signed the binaries. Tested with both 6.0 and the latest builds, and none are signed.

### Expected Behavior
All native binaries in gradle*-bin.zip should be signed.

### Current Behavior
None of the native binaries are signed.

**Comment by ThadHouse** [comment]
The following is the list of binaries that fail in at least 6.0. Could be different in the latest builds.

```
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/native-platform-osx-amd64-0.18.jar/net/rubygrapefruit/platform/osx-amd64/libnative-platform.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/native-platform-osx-amd64-0.18.jar/net/rubygrapefruit/platform/osx-amd64/libnative-platform.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/native-platform-osx-amd64-0.18.jar/net/rubygrapefruit/platform/osx-amd64/libnative-platform-curses.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/native-platform-osx-amd64-0.18.jar/net/rubygrapefruit/platform/osx-amd64/libnative-platform-curses.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/kotlin-compiler-embeddable-1.3.50-patched-for-gradle-6.0.1.jar/META-INF/native/osx/libjansi.jnilib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "WPILibInstaller.dmg/WPILibInstaller.app/Contents/Resources/WPILib_Mac-2020.3.2-artifacts.tar.gz/WPILib_Mac-2020.3.2-artifacts.tar/installUtils/gradle-6.0.1-bin.zip/gradle-6.0.1/lib/kotlin-compiler-embeddable-1.3.50-patched-for-gradle-6.0.1.jar/META-INF/native/osx/libjansi.jnilib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
```

**Comment by ThadHouse** [comment]
Didn't realize we had already posted #14905 . Sorry
